controllers/nodemailer.js

```diff
@@ -0,0 +1,122 @@
+const createError = require("http-errors");
+const jwt = require("jsonwebtoken");
+
+const nodemailerService = require("../services/nodemailer");
+const transporter = require("../configs/nodemailer");
+
+exports.postSendMail = async (req, res, next) => {
+  const { email } = req.body;
+
+  try {
+    const { memoroomId } = req.params;
+    const token = jwt.sign({ email }, process.env.SECRET_KEY, {
+      expiresIn: "1h",
+    });
+    const url = `${process.env.PATH}/${memoroomId}/${process.env.INVITE_URL}?token=${token}`;
+
+    const message = {
+      from: process.env.GOOGLE_MAIL,
+      to: email,
+      subject: "memona C에서 당신을 초대합니다",
+      html: `<div
+      style='
+      text-align: center; 
+      width: 50%; 
+      height: 60%;
+      margin: 15%;
+      padding: 20px;
+      box-shadow: 1px 1px 3px 0px #999;
+      '>
+      <h2>안녕하세요.</h2>
+      <h2>memona C에서 당신을 초대합니다.</h2>
+      <br />
+      <p>아래 링크를 누르면 초대된 Memo Room으로 이동합니다.</p>
+      <a href=${url}>초대링크</a>
+      </div>`,
+    };
+
+    transporter.sendMail(message, (err) => {
+      if (err) {
+        next(err);
+      } else {
+        res.json({
+          result: "success",
+        });
+      }
+    });
+  } catch (err) {
+    if (err.code === "EENVELOPE") {
+      res.status(400).json({
+        result: "fail",
+        error: {
+          message: "Check the email",
+        },
+      });
+
+      return;
+    }
+
+    next(createError(500, "Invalid Server Error"));
+  }
```

Comment:
```suggestion    
    await transporter.sendMail(message);
    
    res.json({ result: "success" });
    return
  } catch (err) {
    if (err.code === "EENVELOPE") {
      res.status(400).json({
        result: "fail",
        error: {
          message: "Check the email",
        },
      });

      return;
    }
    
    if ("sendEmail 호출하였을때 발생되는 에러") {
      res.status(400).json({
        result: "fail",
        error: {
          message: "메일 발송 실패... "
        }
      });
    }

    return;
   
    next(createError(500, "Invalid Server Error"));
  }
```
콜백함수 말고 async await 를 사용하면 좋을것 같아요!!

위의 예시 코드 처럼 메일을 발송 실패 했을때 발생되는 에러도 잡아서 catch 문에 추가 하시면 좋을 것 같아요!!

Reply:
넵! 한번 수정해보겠습니다!